Jamfile

```diff
@@ -376,6 +377,12 @@ rule building ( properties * )
 		}
 	}
 
+	if [ configure.builds /boost/architecture//x86 : $(properties) : x86 ]
+		&& ! <target-os>windows in $(properties)
```

Comment:
target-os doesn't look right here. shouldn't it check toolset gcc? is -mpopcnt a gcc option?